spec/concurrent/edge/future_spec.rb

```diff
@@ -312,6 +312,16 @@
     it 'has flat map' do
       f = Concurrent.future { Concurrent.future { 1 } }.flat.then(&:succ)
       expect(f.value!).to eq 2
+
+      err = StandardError.new('boo')
+      f   = Concurrent.future { Concurrent.failed_future(err) }.flat
+      expect(f.reason).to eq err
+
+      f   = Concurrent.future { raise 'boo' }.flat
```

Comment:
I think this should be `Concurrent.future { Concurrent.future { raise 'boo' } }.flat` since `#flat` is expecting to be returned a future. (Or add this case as an additional test)